mysql-test/lib/mtr_report.pm

```diff
@@ -524,17 +524,13 @@ sub mtr_report_stats ($$$$) {
       $xml_report .= qq(\t\t<testcase assertions="" classname="$current_suite" name="$test->{'name'}" ).
                      qq(status="$test_result" time="$test_time" combinations="$combinations");
 
-      my $comment = $test->{'comment'};
-      $comment =~ s/[\"]//g;
-      $comment =~ s/&/&#38;/g;
-      $comment =~ s/'/&#39;/g;
-      $comment =~ s/"/&#34;/g;
-      $comment =~ s/</&lt;/g;
-      $comment =~ s/>/&gt;/g;
+      my $warnings= replace_special_symbols($test->{'warnings'});
+      my $comment= replace_special_symbols($test->{'comment'});
 
       if ($test->{'result'} eq "MTR_RES_FAILED") {
         my $logcontents = $test->{'logfile-failed'} || $test->{'logfile'};
-        $logcontents= $logcontents.$test->{'warnings'}."\n".$test->{'comment'};
+        $logcontents= replace_special_symbols($logcontents);
```

Comment:
Are you sure, that's correct? `$logcontent` is inside CDATA it doesn't need xml entities, they won't be decoded there by the parser.

Reply:
Removed the special character replacement for `CDATA`. `CDATA` works well without replacement.